src/API.jl

```diff
@@ -894,15 +894,33 @@ end
 
 precompile() = precompile(Context())
 function precompile(ctx::Context)
+    
+    is_stdlib_from_name(name::String) = name in values(stdlibs())
+    
     printpkgstyle(ctx, :Precompiling, "project...")
+    
+    num_tasks = parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS + 1)))
+    parallel_limiter = Channel{Bool}(num_tasks)
```

Comment:
Would a `Base.Semaphore` be more idiomatic?
Conceptual it is for this, right?

It would be thrrad-safe also.
Which might introduce some overhead?
(Idk how hard it is to instantiate a `Semaphore` with a nonthreadsafe `Condition`, i think it can be done?)
But that overhead is probably tiny, and it would save on allocations.

Reply:
Yeah, seems to be a better fit, although I actually see a slight increase in allocations
```
 80.416509 seconds (1.07 M allocations: 88.681 MiB, 0.04% gc time)
```
I guess using the right tooling is the best approach, and if unnecessary allocations are occurring they can be fixed in base.
I'll push the update, but let me know if you think the allocations increase is worth a revert